modules/apps/translation/translation-web/src/main/java/com/liferay/translation/web/internal/portlet/action/ImportTranslationMVCActionCommand.java

```diff
@@ -223,25 +237,90 @@ private void _checkPermission(
 
 	private void _importXLIFFFile(
 			ActionRequest actionRequest, long groupId, String className,
-			long classPK, InputStream inputStream)
+			long classPK, Map<String, String> importErrors,
+			List<String> importSuccess, InputStream inputStream,
+			String uploadedFileName)
 		throws IOException, PortalException {
 
-		TranslationSnapshot translationSnapshot =
-			_translationSnapshotProvider.getTranslationSnapshot(
-				groupId, new InfoItemReference(className, classPK),
-				inputStream);
-
-		_translationEntryService.addOrUpdateTranslationEntry(
-			groupId,
-			_language.getLanguageId(translationSnapshot.getTargetLocale()),
-			new InfoItemReference(className, classPK),
-			translationSnapshot.getInfoItemFieldValues(),
-			ServiceContextFactory.getInstance(actionRequest));
+		try {
+			TranslationSnapshot translationSnapshot =
+				_translationSnapshotProvider.getTranslationSnapshot(
+					groupId, new InfoItemReference(className, classPK),
+					inputStream);
+
+			_translationEntryService.addOrUpdateTranslationEntry(
+				groupId,
+				_language.getLanguageId(translationSnapshot.getTargetLocale()),
+				new InfoItemReference(className, classPK),
+				translationSnapshot.getInfoItemFieldValues(),
+				ServiceContextFactory.getInstance(actionRequest));
+
+			importSuccess.add(uploadedFileName);
+		}
+		catch (XLIFFFileException xliffFileException) {
```

Comment:
Can you move the try/catch block to the caller? (inside the loop) This will make it unnecessary to pass in the file name and collections.

Also, instead of a gigantic if/then/else chain, it would be better to create a static Map<Class<?>, String> to map each exception class to its message. This would reduce the amount of repetitive code needed.